test/hack/bin/check-runbooks.sh

```diff
@@ -90,100 +98,59 @@ main() {
-            if ! isInArray "$runbook" "${runbooks[@]}" && [[ "$runbook" != http* ]]; then
-                local message="file $prettyRulesFilename / alert \"$alertname\" links to unexisting runbook (\"$runbook\")"
+            # Check if url is in runbooks array
+            if ! isInArray "$url" "${runbooks[@]}"; then
+                local message="File $filename:$line_number links to nonexisting URL $url"
                 E_unexistingrunbook+=("$message")
+                # Store annotation data as pipe-separated values for JSON generation
+                annotations_data+=("$filename|$line_number|$url")
                 continue
             fi
+        done <<< "$urls"
+    done
 
-            if [[ "$DEBUG_MODE" != "false" ]]; then
-                echo "file $prettyRulesFilename / alert: $alertname / runbook: $runbook - OK"
-            fi
-
-        # parse rules yaml files, and for each rule found output alertname, runbook, and severity, space-separated, on one line.
-        done < <("$GIT_WORKDIR/$YQ" -o json "$rulesFile" | "$GIT_WORKDIR/$JQ" -j '.spec.groups[]?.rules[] | .alert, " ", .annotations.runbook_url, " ", .labels.severity, "\n"')
-
-        checkedRules+=("$rulesFile")
-    done < <(find "${RULES_FILES[@]}" -type f -print0)
-
-
-    # Output section - let's write down our findings
-    #################################################
-
-    if [[ "$CHECK_EXTRADATA_ERRORS" != "false" ]]; then
-        if [[ "${#E_extradata[@]}" -gt 0 ]]; then
-            echo ""
-            echo "Alerts that failed parsing: ${#E_extradata[@]}"
-            for message in "${E_extradata[@]}"; do
-                echo "$message"
-            done
-            returncode=1
-        fi
-    fi
-    if [[ "$CHECK_NORUNBOOK_ERRORS" != "false" ]]; then
-        if [[ "${#E_norunbook[@]}" -gt 0 ]]; then
-            echo ""
-            echo "Alerts missing runbook: ${#E_norunbook[@]}"
-            for message in "${E_norunbook[@]}"; do
-                echo "$message"
-            done
-            returncode=1
-        fi
-    fi
-    if [[ "$CHECK_UNEXISTINGRUNBOOK_ERRORS" != "false" ]]; then
-        if [[ "${#E_unexistingrunbook[@]}" -gt 0 ]]; then
-            echo ""
-            echo "Alerts using unexisting runbook: ${#E_unexistingrunbook[@]}"
-            for message in "${E_unexistingrunbook[@]}"; do
-                echo "$message"
-            done
-            returncode=1
-        fi
+    if [[ "${#E_unexistingrunbook[@]}" -gt 0 ]]; then
+        echo ""
+        echo "${#E_unexistingrunbook[@]} bad runbook URLs found" >> $GITHUB_STEP_SUMMARY
+        for message in "${E_unexistingrunbook[@]}"; do
+            echo "$message"
+        done
+        
+        # Generate GitHub annotations JSON file
+        echo ""
+        echo "Writing GitHub annotations to /home/runner/work/prometheus-rules/annotations.json"
+        generateAnnotationsJson "${annotations_data[@]}" > /home/runner/work/prometheus-rules/annotations.json
```

Comment:
`/home/runner/work/prometheus-rules/annotations.json` may not be a valid path when running locally.
Maybe writing annotations should be if `if [[ "$runInCi" != false ]];` ?

Reply:
I'm changing to only execute this if `$GITHUB_ENV` is set.